File: Arrays/1572. Matrix Diagonal Sum.js
link = 'https://leetcode.com/problems/matrix-diagonal-sum/'


// Solution
//---------------------------------
const diagonalSum = (mat) => {
    let elements = []
    let len = mat.length
    if(len===1){
        return mat[0][0]
    }
    for (let i = 0; i < len; i++) {
        if(i!==len-1-i){
            elements.push(mat[i][i])
            elements.push(mat[i][len-1-i])
        }
    }
    if(len%2!==0){
        elements.push(mat[Math.floor(len/2)][Math.floor(len/2)])
    }
    let sum =  elements.reduce((accumulator, item)=>{
        return accumulator + item
    },0)
    return sum
}
//---------------------------------
let result = diagonalSum([[1,2,3],
    [4,5,6],
    [7,8,9]])
console.log(result)

File: Arrays/1572. Matrix Diagonal Sum.py
def diagonalSum(mat):
    elements = []
    length = len(mat)
    if length==1:
        return mat[0][0]
    for i in range(length):
        if i!=length-1-i:
            elements.append(mat[i][i])
            elements.append(mat[i][length-1-i])
    if length%2!=0:
        elements.append(mat[length//2][length//2])

    sumation = sum(elements)
    return sumation

result = diagonalSum([[1,2,3],
    [4,5,6],
    [7,8,9]])

print(result)